src/DataSourcePoller.php

```diff
@@ -0,0 +1,231 @@
+<?php
+
+namespace Automattic\WooCommerce\Admin;
+
+/**
+ * Specs data source poller class.
+ * This handles polling specs from JSON endpoints, and
+ * stores the specs in to the database as an option.
+ */
+class DataSourcePoller {
+
+	/**
+	 * Name of data sources filter.
+	 */
+	const FILTER_NAME = 'data_source_poller_data_sources';
+
+	/**
+	 * Default data sources array.
+	 *
+	 * @var array
+	 */
+	protected $data_sources = array();
+
+	/**
+	 * Key for the spec id.
+	 *
+	 * @var string
+	 */
+	protected $spec_key = 'id';
+
+	/**
+	 * Key for the spec id.
+	 *
+	 * @var string
+	 */
+	protected $transient_name = '';
+
+	/**
+	 * The logger instance.
+	 *
+	 * @var WC_Logger|null
+	 */
+	protected static $logger = null;
+
+	/**
+	 * Constructor.
+	 *
+	 * @param array  $data_sources urls for data sources.
+	 * @param string $transient_name name of transient.
+	 * @param string $spec_key Optional key used as the spec identifier.
+	 */
+	public function __construct( $data_sources, $transient_name, $spec_key = 'id' ) {
```

Comment:
If we do keep the generic class, it would be great to convert this to a single argument `$args` with an associative array of keys, otherwise this argument list could grow pretty lengthy.

Reply:
I like the generic `$args` prop